From d39840541ea7dce77b25fdb3c25e30c70f02874c Mon Sep 17 00:00:00 2001
From: Saikiran <bjsaikiran@gmail.com>
Date: Sun, 25 Jan 2026 11:13:10 +0530
Subject: [PATCH 1/3] Revert "arm64: dts: qcom: x1-el2: Add qcom,broken-reset
 for remoteprocs"

This reverts commit 408fdf8a8d41b35bbd5dffbb1a888056354fe8cf.

The original commit introduced "qcom,broken-reset" to skip ADSP firmware
loading and instead attach to the pre-boot state. This was intended for EL2
(Hypervisor) environments. However, on bare-metal configurations (like the
Lenovo Yoga Slim 7x), this prevents the necessary Linux-specific ADSP firmware
(qcadsp8380.mbn) from loading.
By skipping the reset/load phase, the ADSP remains in a "lite" state managed
by the bootloader, which is insufficient for full audio subsystem initialization
in Linux, causing sound card registration failures (error -2 on firmware load,
followed by a non-functional attachment).
Reverting this allows the remoteproc driver to properly reset the hardware
and load the correct firmware during boot.

Signed-off-by: Saikiran <bjsaikiran@gmail.com>
---
 arch/arm64/boot/dts/qcom/x1-el2.dtso | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/arch/arm64/boot/dts/qcom/x1-el2.dtso b/arch/arm64/boot/dts/qcom/x1-el2.dtso
index bc36b2e1bb7f..af98c2514c4b 100644
--- a/arch/arm64/boot/dts/qcom/x1-el2.dtso
+++ b/arch/arm64/boot/dts/qcom/x1-el2.dtso
@@ -58,14 +58,6 @@ &pcie_smmu {
 	status = "okay";
 };
 
-&remoteproc_adsp {
-	qcom,broken-reset;
-};
-
-&remoteproc_cdsp {
-	qcom,broken-reset;
-};
-
 /*
  * The "SBSA watchdog" is implemented in software in Gunyah
  * and can't be used when running in EL2.
-- 
2.51.0

